### PR TITLE
Remove default empty state of `sf::Music`

### PR DESCRIPTION
**Pull request by vittorioromeo**
Same philosophy as

- #2992 
- #3020 
- #3021 
- #3022
- #3025
- #3028
- #3030
- #3032
- #3027

Note: the sound effects example will be simplified like #3062 and #3071 in a subsequent PR.

### CODE REVIEW

**Comment by vittorioromeo** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Audio/Music.cpp [hunk 1]
```diff
@@ -39,80 +40,89 @@
...
+}
 
-    // Perform common initializations
-    initialize();
 
-    return true;
+////////////////////////////////////////////////////////////
+std::optional<Music> Music::openFromFile(const std::filesystem::path& filename)
+{
+    return tryOpenFromFile(sf::InputSoundFile::openFromFile(filename));
```
Comment (ChrisThrasher) [inline src/SFML/Audio/Music.cpp:97]:
```suggestion
    return tryOpenFromFile(InputSoundFile::openFromFile(filename));
```
Similar comment applies elsewhere in the file.

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]
I think this generally looks good. Glad to make music moveable and wrap up (I think?) the last of the `std::optional` factory functions.

File: include/SFML/Audio/AudioResource.hpp [hunk 2]
```diff
@@ -47,11 +47,36 @@ class SFML_AUDIO_API AudioResource
     ////////////////////////////////////////////////////////////
     AudioResource();
 
+public:
```
[the remark this answers is not in the document]
Comment (vittorioromeo) [reply include/SFML/Audio/AudioResource.hpp:50]: Will change

File: src/SFML/Audio/Music.cpp (hunk 1 above)
Comment (vittorioromeo) [reply src/SFML/Audio/Music.cpp:97]: Will change

File: src/SFML/Audio/Music.cpp [hunk 3]
```diff
@@ -179,80 +189,71 @@ void Music::setLoopPoints(TimeSpan timePoints)
 ////////////////////////////////////////////////////////////
 bool Music::onGetData(SoundStream::Chunk& data)
 {
-    assert(m_file && "Music::onGetData() Cannot perform operation until music is opened");
```
[the remark this answers is not in the document]
Comment (vittorioromeo) [reply src/SFML/Audio/Music.cpp:182]: Will remove

File: test/Audio/Music.test.cpp [hunk 4]
```diff
@@ -34,39 +32,17 @@ TEST_CASE("[Audio] sf::Music", runAudioDeviceTests())
...
-            const auto [offset, length] = music.getLoopPoints();
-            CHECK(offset == sf::Time::Zero);
-            CHECK(length == sf::Time::Zero);
-            CHECK(music.getChannelCount() == 0);
-            CHECK(music.getSampleRate() == 0);
-            CHECK(music.getStatus() == sf::Music::Status::Stopped);
-            CHECK(music.getPlayingOffset() == sf::Time::Zero);
-            CHECK(!music.getLoop());
+            auto music = sf::Music::openFromFile("does/not/exist.wav");
+            REQUIRE(!music.has_value());
```
[the remark this answers is not in the document]
Comment (vittorioromeo) [reply test/Audio/Music.test.cpp:40]: Will change

File: test/Audio/Music.test.cpp [hunk 5]
```diff
@@ -34,39 +32,17 @@ TEST_CASE("[Audio] sf::Music", runAudioDeviceTests())
...
         }
 
         SECTION("Valid file")
         {
-            REQUIRE(music.openFromFile("Audio/ding.mp3"));
+            auto music = sf::Music::openFromFile("Audio/ding.mp3").value();
```
[the remark this answers is not in the document]
Comment (vittorioromeo) [reply test/Audio/Music.test.cpp:45]: Good point, will change

File: test/Audio/Music.test.cpp [hunk 6]
```diff
@@ -82,25 +58,18 @@ TEST_CASE("[Audio] sf::Music", runAudioDeviceTests())
...
+            auto music = sf::Music::openFromMemory(memory.data(), memory.size());
+            REQUIRE(!music.has_value());
         }
 
         SECTION("Valid buffer")
         {
             memory = loadIntoMemory("Audio/ding.flac");
-            REQUIRE(music.openFromMemory(memory.data(), memory.size()));
+
+            auto music = sf::Music::openFromMemory(memory.data(), memory.size()).value();
```
[the remark this answers is not in the document]
Comment (vittorioromeo) [reply test/Audio/Music.test.cpp:72]: Will change

File: test/Audio/Music.test.cpp [hunk 7]
```diff
@@ -116,25 +85,18 @@ TEST_CASE("[Audio] sf::Music", runAudioDeviceTests())
...
+            auto music = sf::Music::openFromStream(stream);
+            REQUIRE(!music.has_value());
         }
 
         SECTION("Valid stream")
         {
             REQUIRE(stream.open("Audio/doodle_pop.ogg"));
-            REQUIRE(music.openFromStream(stream));
+
+            auto music = sf::Music::openFromStream(stream).value();
```
[the remark this answers is not in the document]
Comment (vittorioromeo) [reply test/Audio/Music.test.cpp:99]: Will change

**Comment by vittorioromeo** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Audio/Music.cpp (hunk 1 above)
Comment (ChrisThrasher) [reply src/SFML/Audio/Music.cpp:95]: I think you missed this

**Review by ChrisThrasher — APPROVED** [review]

**Comment by vittorioromeo** [comment]
SFMLBuildMaster: Build this please.

**Pull request merged**